### PR TITLE
Tests(Qtm-323)Migrating Badge, Button,Card and AutoComplete Unit tests

### PR DESCRIPTION
**Pull request by matheusrpp**
## Description
https://jirasoftware.catho.com.br/browse/QTM-323 + AutoComplete 

I added the user-event package to latter improve AutoComplete tests with arrow navigation tests.

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Code review

### CODE REVIEW

File: components/Badge/Badge.unit.test.jsx [hunk 1]
```diff
@@ -1,31 +1,59 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom';
```
Comment (toncabral88) [inline components/Badge/Badge.unit.test.jsx:3]: How about include this import only in jest-config.js?

File: components/AutoComplete/AutoComplete.unit.test.jsx [hunk 2]
```diff
@@ -1,116 +1,151 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
-import { mount } from 'enzyme';
+import { fireEvent, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
```
Comment (toncabral88) [inline components/AutoComplete/AutoComplete.unit.test.jsx:3]: How about include this import only in jest-config.js?

File: components/AutoComplete/AutoComplete.unit.test.jsx [hunk 3]
```diff
@@ -1,116 +1,151 @@
...
+import userEvent from '@testing-library/user-event';
```
Comment (toncabral88) [inline components/AutoComplete/AutoComplete.unit.test.jsx:4]: This is nice!

File: components/Button/Button.unit.test.jsx [hunk 4]
```diff
@@ -1,79 +1,133 @@
 import React from 'react';
-import renderer from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
 import Button from './Button';
+import '@testing-library/jest-dom';
```
Comment (toncabral88) [inline components/Button/Button.unit.test.jsx:4]: How about include this import only in jest-config.js?

**Review by toncabral88 — APPROVED** [review]
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Code review

How about we remove coverage folder from gitignore? I think when the values change on coverage-final.json will be easy to notice the differences.

**Review by rapahaeru — DISMISSED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Code review

**Review by toncabral88 — APPROVED** [review]

**Review by rapahaeru — APPROVED** [review]

**Pull request merged**